01_fib/joshcarp/fib_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var stdout = os.Stdout
+var tests = map[int]string{7: "1\n1\n2\n3\n5\n8\n13\n",
+	-7: "1\n-1\n2\n-3\n5\n-8\n13\n"}
+
```

Comment:
Use table test to add more test cases. Refer to https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go

Reply:
interesting, I would say that the map method is nicer than using a struct in this case;
i feel like this has added complexity:
```
var tests = map[string]struct {
	input    int
	expected string
}{
	"positive": {7, "1\n1\n2\n3\n5\n8\n13\n"},
	"negative": {-7, "1\n-1\n2\n-3\n5\n-8\n13\n"},
}
```

Reply:
I guess the point is that usually you declare the testcases (map or slice) local to a test function and iterate over them rather than defining two or more different functions.
Here is a more recent dave cheney post that illustrates this including sub-tests:
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests